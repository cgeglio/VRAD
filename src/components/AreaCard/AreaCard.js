import React from 'react';
import PropTypes from 'prop-types';
import './AreaCard.scss';
import { Link } from 'react-router-dom';

export const AreaCard = ({ id, nickname, longname, description, listings, addListingsToState }) => {
  return (
    <div className="area-card">
      <img src={process.env.PUBLIC_URL + `/${nickname.split(' ').join('')}.jpg`} alt="Area" />
      <h2>{nickname}</h2>
      <span>{longname}</span>
      <p>{description}</p>
      <Link to={`/area/${id}/listings`}>
        <button type="button" className="view-listings" onClick={() => addListingsToState(listings)}>View Listings</button>
      </Link>
    </div>
  );
};

AreaCard.propTypes = {
  id: PropTypes.number.isRequired,
  nickname: PropTypes.string.isRequired,
  longname: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  listings: PropTypes.array,
  addListingsToState: PropTypes.func,
};
