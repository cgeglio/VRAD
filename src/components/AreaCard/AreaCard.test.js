import React from 'react';
import { shallow } from 'enzyme';
import { AreaCard } from './AreaCard';

describe('AreaCard', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockArea = {
      nickname: 'test nickname',
      longname: 'test longname',
      description: 'test description',
      id: 5,
      listings: [{ name: 'test name', listing_id: 5 }],
    };
    const mockAddListingsToState = jest.fn();

    const wrapper = shallow(<AreaCard
      {...mockArea}
      key={mockArea.id}
      addListingsToState={mockAddListingsToState}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
