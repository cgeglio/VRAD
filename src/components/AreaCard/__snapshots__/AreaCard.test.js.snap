// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaCard should match the snapshot with all data passed in correctly 1`] = `
<div
  className="area-card"
>
  <img
    alt="Area"
    src="/testnickname.jpg"
  />
  <h2>
    test nickname
  </h2>
  <span>
    test longname
  </span>
  <p>
    test description
  </p>
  <Link
    to="/area/5/listings"
  >
    <button
      className="view-listings"
      onClick={[Function]}
      type="button"
    >
      View Listings
    </button>
  </Link>
</div>
`;
